pipelines/zendesk/helpers/api_helpers.py

```diff
@@ -0,0 +1,100 @@
+        )
+    elif isinstance(credentials, ZendeskCredentialsEmailPass):
+        zendesk_client = Zenpy(
+            email=credentials.email, subdomain=credentials.subdomain, password=credentials.password,
+            domain=domain, timeout=timeout, ratelimit_budget=ratelimit_budget, proactive_ratelimit=proactive_ratelimit, proactive_ratelimit_request_interval=proactive_ratelimit_request_interval,
+            disable_cache=disable_cache
+        )
+    return zendesk_client
+
+
+def process_ticket(ticket: Ticket, custom_fields: DictStrStr, pivot_fields: bool = True) -> DictStrAny:
+    """
+    Helper that returns a dictionary of the ticket class provided as a parameter. This is done since to_dict()
+    method of Ticket class doesn't return all the required information
+    @:param sample_ticket: Ticket Object returned by a Zenpy API call, contains info on a single ticket.
+    @:param custom_fields: A dict containing all the custom fields available for tickets.
+    @:param pivot_fields: Bool that indicates whether to pivot all custom fields or not.
+    @:return base_dict: A dict containing 'cleaned' data about a ticket.
+    """
+
+    base_dict: DictStrAny = ticket.to_dict()
+    # get tags as a string eliminating the square brackets from string
+    base_dict["tags"] = str(base_dict["tags"])[1:-1]
+
+    # pivot custom field if indicated as such
+    # get custom fields
+    for custom_field in base_dict["custom_fields"]:
+        if pivot_fields:
+            cus_field_id = str(custom_field["id"])
+            field_name = custom_fields[cus_field_id]
+            base_dict[field_name] = custom_field["value"]
+        else:
+            custom_field["ticket_id"] = ticket.id
+
+    # delete fields that are not needed for pivoting
+    if pivot_fields:
+        del base_dict["custom_fields"]
+    del base_dict["fields"]
+
+    # modify dates to return datetime objects instead
+    base_dict["updated_at"] = ticket.updated
+    base_dict["created_at"] = ticket.created
+    base_dict["due_at"] = ticket.due
+    return base_dict
+
+
+def basic_load(resource_api: Iterator[Any]) -> Iterator[TDataItem]:
+    """
+    Receives a generator/iterable of Zenpy Objects and returns a generator of dicts. Loader helper
+    @:param resource_api: generator/iterable of Zenpy Objects
+    """
+    # sometimes there is a single element which is a dict instead of a generator of objects
+    if isinstance(resource_api, dict):
+        yield resource_api
+    else:
+        # some basic resources return a dict instead of objects
+        for element in resource_api:
+            if isinstance(element, dict):
+                yield element
+                continue
+            try:
+                dict_res = element.to_dict()
+                yield dict_res
+            except AttributeError as e:
+                logger.warning(f"Error! Element {element} of type {type(element)} returned from api {resource_api} has no attribute to_dict")
```

Comment:
this looks very dangerous. aren't we expecting this method in all cases?

Reply:
we are, it's fixed now